File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        char_indexes = {}
        for i, ch in enumerate(s):
            if ch in char_indexes:
                char_indexes[ch].append(i)
            else:
                char_indexes[ch] = [i]
        
        min_i = len(s) + 1
        for ch, arr in char_indexes.items():
            if len(arr) == 1:
                if arr[0] < min_i:
                    min_i = arr[0]
        
        if min_i == len(s) + 1:
            return -1
        return min_i
        

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int[] letters = new int[26];
        for (int i = 0; i < s.length(); i++) {
            letters[s.charAt(i)-'a']++;
        }
        boolean noAnswer = true;
        for (int i = 0; i < 26; i++) {
            if (letters[i] == 1) {
                noAnswer = false;
                break;
            }
        }
        if (noAnswer) {
            return -1;
        }
        for (int i = 0; i < s.length(); i++) {
            if (letters[s.charAt(i)-'a'] == 1) {
                return i;
            }
        }
        return -1;
    }
}